sdk/webpubsub/azure-messaging-webpubsubservice/azure/messaging/webpubsubservice/_patch.py

```diff
@@ -0,0 +1,327 @@
+        "token": token,
+        "url": "{}?access_token={}".format(client_url, token),
+    }
+
+
+class JwtCredentialPolicy(SansIOHTTPPolicy):
+
+    NAME_CLAIM_TYPE = "http://schemas.xmlsoap.org/ws/2005/05/identity/claims/name"
+
+    def __init__(self, credential, user=None):
+        # type: (AzureKeyCredential, typing.Optional[str]) -> None
+        """Create a new instance of the policy associated with the given credential.
+
+        :param credential: The azure.core.credentials.AzureKeyCredential instance to use
+        :type credential: ~azure.core.credentials.AzureKeyCredential
+        :param user: Optional user name associated with the credential.
+        :type user: str
+        """
+        self._credential = credential
+        self._user = user
+
+    def on_request(self, request):
+        # type: (PipelineRequest) -> typing.Union[None, typing.Awaitable[None]]
+        """Is executed before sending the request from next policy.
+
+        :param request: Request to be modified before sent from next policy.
+        :type request: ~azure.core.pipeline.PipelineRequest
+        """
+        request.http_request.headers["Authorization"] = "Bearer " + self._encode(
+            request.http_request.url
+        )
+        return super(JwtCredentialPolicy, self).on_request(request)
+
+    def _encode(self, url):
+        # type: (AzureKeyCredential) -> str
+        data = {
+            "aud": url,
+            "exp": datetime.now(tz=_UTC_TZ()) + timedelta(seconds=60),
+        }
+        if self._user:
+            data[self.NAME_CLAIM_TYPE] = self._user
+
+        encoded = jwt.encode(
+            payload=data,
+            key=self._credential.key,
+            algorithm="HS256",
+        )
+        return six.ensure_str(encoded)
+
+
+class ApiManagementProxy(CustomHookPolicy):
+
+    def __init__(self, **kwargs):
+        # type: (typing.Optional[str], typing.Optional[str]) -> None
+        """Create a new instance of the policy.
+
+        :param endpoint: endpoint to be replaced
+        :type endpoint: str
+        :param proxy_endpoint: proxy endpoint
+        :type proxy_endpoint: str
+        """
+        self._endpoint = kwargs.pop('origin_endpoint', None)
+        self._reverse_proxy_endpoint = kwargs.pop('reverse_proxy_endpoint', None)
+        super(ApiManagementProxy, self).__init__(**kwargs)
+
+    def on_request(self, request):
+        # type: (PipelineRequest) -> None
+        """Is executed before sending the request from next policy.
+
+        :param request: Request to be modified before sent from next policy.
+        :type request: ~azure.core.pipeline.PipelineRequest
+        """
+        super(ApiManagementProxy, self).on_request(request)
+        if self._endpoint and self._reverse_proxy_endpoint:
+            request.http_request.url = request.http_request.url.replace(self._endpoint, self._reverse_proxy_endpoint)
+
+
+class WebPubSubServiceClientConfiguration(Configuration):
```

Comment:
I think we'd better change auto-generated code. The only place we could change is `_patch.py`. Otherwise we should remember every place we ever changed. This will gradually become maintenance disaster.